wolfcrypt/src/ecc.c

```diff
@@ -4036,6 +4130,16 @@ int wc_ecc_verify_hash_ex(mp_int *r, mp_int *s, const byte* hash,
 
 #else
 
+  /* checking if private key with no public part */
+  if (key->type == ECC_PRIVATEKEY_ONLY) {
+      WOLFSSL_MSG("Verify called with private key, generating public part");
+      err = wc_ecc_make_pub_ex(key, NULL, NULL);
+      if (err != MP_OKAY) {
+           WOLFSSL_MSG("Unable to extract public key");
+           return BAD_FUNC_ARG;
```

Comment:
The thought process there was that it was a bad function argument if a bad key was passed in. I agree that it should be changed to returning err though. That will be more informative to users about the reason for encountering an error. Thanks, will push a commit to fix it.